src/index.ts

```diff
@@ -35,10 +35,15 @@ function createPlugin(
   const mdxPlugin: MdxPlugin = {
     name: 'vite-plugin-mdx',
     // I can't think of any reason why a plugin would need to run before mdx; let's make sure `vite-plugin-mdx` runs first.
-    enforce: "pre",
+    enforce: 'pre',
     mdxOptions: globalMdxOptions,
     configResolved({ root, plugins }) {
-      const reactRefresh = plugins.find((p) => p.name === 'react-refresh')
+      // @vitejs/plugin-react-refresh has been upgraded to @vitejs/plugin-react,
+      // and the name of the plugin performing `transform` has been changed from 'react-refresh' to 'vite:react-babel',
+      // to be compatible, we need to look for both plugin name.
+      const reactRefresh = plugins.find(
+        (p) => p.name === 'react-refresh' || p.name === 'vite:react-babel'
+      )
```

Comment:
```suggestion
      // We should also look for the other plugins names exported from @vitejs/plugin-react in case there are some internal refactors.
      const reactRefreshPlugins = plugins.filter(
        (p) => p.name === 'react-refresh' || p.name === 'vite:react-babel'
          || p.name === 'vite:react-refresh' || p.name === 'vite:react-jsx'
      );
      const reactRefresh = reactRefreshPlugins.find(p => p.transform);
```
This would also work